### PR TITLE
CP-15745: Add utility scripts for building packages under travis

### PR DESCRIPTION
**Pull request by johnelse**

### CODE REVIEW

**Comment by simonjbeaumont** [comment]
discussed a different way of doing this to keep as much logic as possible within this repo so consumers needn't do any config or duplicate any scripts.

**Comment by simonjbeaumont** [comment]
Also, we may want to consider tagging this repo when you have some scripts that are stable so that the adventurous consumer can wget from master but the cautious can get a stable version. Might enable us to be more free when hacking in this repo without fear of breaking "all the things".

**Comment by johnelse** [comment]
Indeed. I have a hardcoded wget of the master branch at the moment - would be good to parameterise that too.

**Comment by johnelse** [comment]
Removed the hardcoding - it's now possible to override `BUILDENV_USER` and `BUILDENV_BRANCH` to get a different version of the build scripts (which incidentally is how I've been testing this).

File: utils/travis-build-repo.sh [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+# $BUILDENV_BRANCH (optional - default 'master')
+# - The github branch from which to pull the required scripts.
+# $REPO_PACKAGE_NAME (required)
+# - The package for which dependencies will be installed in the container.
+# $REPO_CONFIGURE_CMD (optional - default './configure')
+# - The configure command to run in the repo.
+# $REPO_BUILD_CMD (optional - default 'make')
+# - The build command to run in the repo.
+# $REPO_TEST_CMD (optional - default 'make test')
+# - The test command to run in the repo.
```
Comment (simonjbeaumont) [inline utils/travis-build-repo.sh:20]: Nice docs :+1:

**Comment by simonjbeaumont** [comment]
Looks good! This is gonna be a great step forward.

**Pull request merged**